modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/ArrayRowHandler.java

```diff
@@ -50,6 +52,18 @@ public Object[] concat(Object[] left, Object[] right) {
         return ArrayUtils.concat(left, right);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public Object[] map(Object[] row, int[] mapping, int offset) {
```

Comment:
can we gather new row without new allocations ? just using incoming row and cut all unused stuff at the end ?

Reply:
Yes we can, but this will go against the description of the method in the javadoc. This implementation will only be used in tests. I'm not sure how much can we gain from such optimization. But if you insist, I can change this method to return the original array with the values replaced.